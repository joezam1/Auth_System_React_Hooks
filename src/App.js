import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import './scss/styleImporter.scss';
import WindowEventManager from './middleware/WindowEventsManager';

import RouteConfig from '../configuration/routes/RouteConfig.js';
import Home from './webPages/public/Home.js';
import Register from './webPages/public/Register.js';
import Login from './webPages/public/Login.js';
import Logout from './webPages/public/Logout.js';
import CustomerDashboard from './webPages/private/CustomerDashboard';
import CustomerOrders from './webPages/private/CustomerOrders';

import ComponentConfig from '../configuration/components/ComponentConfig';


//Test:DONE
export default function App() {

    //======================================================
    console.log('Home', Home);
    console.log('Register', Register());
    console.log('<Register/>', <Register/>);
    let registerComponent = <Register/>
    let registerName = registerComponent.type.name;
    console.log('componentName-<Register/>-registerName', registerName )

    let loginComponent = <Login/>;
    let loginComponentName = loginComponent.type.name;
    console.log('Login-function-Name', loginComponentName);
    //======================================================

    let linkButton = ComponentConfig._LinkButtonPrivateRedirect;
    let linkButtonName = linkButton.type.name;
    console.log('TEST-linkButtonName:', linkButtonName);

     //======================================================


    WindowEventManager.resolveWindowNavigationEvent();

    return (<div className="app">
        <BrowserRouter>
            <Routes>
                <Route exact path={RouteConfig.home} element={<Home />} />
                <Route exact path={RouteConfig.authRegisterPath} element={<Register/>}/>
                <Route exact path={RouteConfig.authLoginPath} element={<Login/>} />
                <Route exact path={RouteConfig.authLogoutPath} element={<Logout/>} />
                <Route exact path={RouteConfig.privateCustomerDashboard} element={<CustomerDashboard/>} />
                <Route exact path={RouteConfig.privateCustomerOrders} element={<CustomerOrders/>} />
            </Routes>
        </BrowserRouter>

    </div>);
}