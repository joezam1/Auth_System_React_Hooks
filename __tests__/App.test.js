import React from "react";
import { render, cleanup } from '@testing-library/react';
import App from '../src/App.js';



describe('File: App.js', function(){
    afterEach(cleanup);
    afterEach(() => {
        jest.clearAllMocks();
    });
    //NOTE: Test that JEST is working correctly
    //test('True is True', function () { expect(true).toBe(true); });

    test('Function App it should take a snapshot', function(){
        var container = render(<App/>);
        expect(container.asFragment(<App/>)).toMatchSnapshot();
    });
});