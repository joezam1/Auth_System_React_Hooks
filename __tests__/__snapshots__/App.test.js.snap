// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File: App.js Function App it should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="home"
    >
      This is home Section - Public Page
      <br />
      <ul>
        <li>
           
          <a
            data-testid="link-home-register-id"
            href="/auth/register"
          >
            Register
          </a>
        </li>
        <li>
           
          <a
            data-testid="link-home-login-id"
            href="/auth/login"
          >
             Login
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
